src/asciidoc/docs/guidelines/persistSessionData.adoc

```diff
@@ -0,0 +1,60 @@
+= Persisting data across sessions
+:page-slug: persist-session-data
+:page-description: How to persist data that doesn't expire with the session.
+
+Data can be stored temporarily using window.sessionStorage.
+However, data stored in this way will expire when the session ends.
+If data needs to be persisted across sessions, `window.localStorage` may be used.
+Data stored in this way has no expiration date.
+It will be available until the browser's cache is cleared.
+
+For example, here is link to a Koji that uses local storage to store the user's name: https://withkoji.com/~Svarog1389/chatbox[Chatbox]
+
+
+When you first run the Koji, your user name is set to Anonymous.
+You can change it by clicking on the cog icon in the top left corner.
+The new name will be saved to local storage.
+The relevant code is in the `Chatbox.js` file of the template: https://withkoji.com/code/Svarog1389/chatbox#frontend%2Fcommon%2FComponents%2FChatbox.js[Chatbox.js]
+
+== Using local storage
+There are two ways to use `window.localStorage`.
+
+Use property names directly:
+[source,javascript]
+----
+localStorage.username = 'johnsmith001';
+console.log(localStorage.username); // prints johnsmith001 to the console
+delete localStorage.username;
+console.log(localStorage.username); // prints undefined to the console
+----
+
+Alternatively, `localStorage` provides the methods `setItem` and `removeItem`, which you can use as follows:
+[source,javascript]
+----
+localStorage.setItem('username', 'johnsmith001');
+console.log(localStorage.getItem('username')); // prints johnsmith001 to the console
+localStorage.removeItem('username');
+console.log(localStorage.getItem('username')); // prints undefined to the console
+----
+
+== Developer tips
+Check that `windows.localStorage` is accessible before using it.
+If the user has selected *Block third-party cookies and site data* in the content settings of the browser, local storage will be inaccessible.
+Local storage may also be inaccessible in incognito mode (private browsing).
+To avoid an error in these situations, always check that local storage is available before using it.
+An easy way to do this is to use something like this:
+[source,javascript]
+----
+function isLocalStorageAvailable() {
+    const test = 'test';
+    try {
+        localStorage.setItem(test, test);
+        localStorage.removeItem(test);
+        return true;
+    } catch (e) {
+        return false;
+    }
+}
+----
+
+If local storage is inaccessible, you can decide to persist the data for only the duration of the session or you can save it to the database (see <<koji-database#, Koji Database>>).
```

Comment:
```suggestion
If local storage is inaccessible, you can decide to persist the data for only the duration of the session or you can save it to a database, such as the built-in <<koji-database#, Koji database>>.
```
Technically, they could use another service, but it might be easier to use the one we provide.

Reply:
Done